controllers/nodenetworkconfigurationpolicy_utils.go

```diff
@@ -0,0 +1,98 @@
+/*
+Copyright The Kubernetes NMState Authors.
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"regexp"
+	"strings"
+)
+
+// The function strips disarranged parts of the error messages, so it is easier to read for the user
+func formatErrorString(err error) string {
+	var sb strings.Builder
+	errorMessage := err.Error()
+	errorLines := strings.Split(errorMessage, "\n")
+	index := 0
+
+	re := regexp.MustCompile(`\s*(.*--no-commit --timeout \d+: 'exit status \d+') '{2}\s'`)
+	matches := re.FindStringSubmatch(errorLines[index])
+	if len(matches) == 2 {
+		sb.WriteString(strings.TrimRight(matches[1], " ") + "\n")
+		index++
+	}
+	index = skipLines(&errorLines, index)
+
+	for index < len(errorLines) {
+		lineSplitByColon := strings.Split(strings.TrimRight(errorLines[index], " "), ": ")
+		indent := ""
+		for _, lineSection := range lineSplitByColon {
+			sb.WriteString(indent + lineSection + "\n")
+			indent = indent + "  "
+		}
+		index++
+		index = skipLines(&errorLines, index)
+	}
+	return sb.String()
+}
+
+func skipLines(lines *[]string, index int) int {
+	if index >= len(*lines) {
+		return index
+	}
+
+	regexMatchFileTrace := `\A\s*File "\S*",\sline\s\d*,\sin`
+	regexTraceback := `\A\s*Traceback\s\(most\srecent\scall\slast\):`
+	regexWithDateTime := `\A\s*\d{4}-\d{2}-\d{2}\s\d{2}:\d{2}:\d{2}`
+	regexCurrentState := `.*->\scurrentState:\s---`
+	regexUnhandledMessage := `\AUnhandled\s.*\sfor\s`
+	regexKeywords := `DEBUG|WARNING|UserWarning:|warnings\.warn\(`
```

Comment:
To be honest I think all this ha to be done at nmstate itself with an output option that shows a simplified error at output and a debug error at a file for example or simplied error at stdout and extensive at stderr.

Doing it at nmstate and adding test will ensure that no new version of nmstate is going to break us.